packages/math/src/pool/concentrated/math.ts

```diff
@@ -270,6 +277,21 @@ export function approxRoot(dec: Dec, root: number, maxIters = 300): Dec {
   return guess;
 }
 
+// monotonicSqrt does a naive 1 ULP shift to the output of approxSqrt to approximate monotonicity.
+// While this function is not technically monotonic, it is close enough for our purposes.
+// True monotonicity can be achieved by replacing the `approxRoot` with a call to big int's lower level
+// square root function (floor sqrt), which in conjunction with the 1 ULP shift will guarantee monotonicity.
+export function monotonicSqrt(dec: Dec, maxIters = 300): Dec {
+  let estimate = approxRoot(dec, 2, maxIters);
+  const inverseValue = estimate.pow(new Int(2));
+
+  if (inverseValue.lt(estimate)) {
+    estimate = estimate.add(smallestDec);
+  }
+
+  return estimate;
+}
```

Comment:
TODO for myself - merge Alpin's PR and switch to using this from a different package

Reply:
done